llvm/lib/ObjectYAML/XCOFFEmitter.cpp

```diff
@@ -486,10 +543,10 @@ bool XCOFFWriter::writeSectionData() {
     XCOFFYAML::Section YamlSec = Obj.Sections[I];
     if (YamlSec.SectionData.binary_size()) {
       // Fill the padding size with zeros.
-      int64_t PaddingSize =
-          InitSections[I].FileOffsetToData - (W.OS.tell() - StartOffset);
+      int64_t PaddingSize = (uint64_t)InitSections[I].FileOffsetToData -
+                            (W.OS.tell() - StartOffset);
       if (PaddingSize < 0) {
-        ErrHandler("redundant data was written before section data");
+        ErrHandler("unexpected data was written before section data");
```

Comment:
Test case?

Reply:
This error (and the 2 similar ones) should be asserts. There's no need to change the wording of the message.